core/lib/zksync_core/src/sync_layer/gossip/conversions.rs

```diff
@@ -36,11 +44,17 @@ impl FetchedBlock {
         let payload = consensus::Payload::decode(&block.payload)
             .context("Failed deserializing block payload")?;
 
+        let protocol_version = block.justification.message.protocol_version;
+        let protocol_version =
+            u16::try_from(protocol_version.as_u32()).context("Protocol version overflow")?;
```

Comment:
nit: u16::try_from error message already includes information about the overflow: https://doc.rust-lang.org/src/core/num/error.rs.html#24

Reply:
>  I don't want to lose PR approvals

I feel your motivation here, but I still hate it :P

Reply:
Since the PR has got merge conflicts, I'll lose approvals anyway, so I've fixed it 👍